plugins/techdocs-addons-contrib/src/ReportIssue/hooks.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import parseGitUrl from 'git-url-parse';
+
+import { useApi } from '@backstage/core-plugin-api';
+import {
+  replaceGitHubUrlType,
+  replaceGitLabUrlType,
+} from '@backstage/integration';
+import { scmIntegrationsApiRef } from '@backstage/integration-react';
+import {
+  useShadowRootElements,
+  useShadowRootSelection,
+} from '@backstage/plugin-techdocs-react';
+
+import { PAGE_EDIT_LINK_SELECTOR } from './constants';
+
+const resolveBlobUrl = (url: string, type: string) => {
+  if (type === 'github') {
+    return replaceGitHubUrlType(url, 'blob');
+  }
+  return replaceGitLabUrlType(url, 'blob');
```

Comment:
Can type be anything else than `gitlab` when we get here? If so what do you think about adding an if statement checking gitlab too, and log something for unsupported types?

Reply:
Good point. Should probably do that!